src/sentry/integrations/api/endpoints/organization_coding_agents.py

```diff
@@ -50,3 +145,217 @@ def get(self, request: Request, organization) -> Response:
+                and "repo_name" in solution_item["relevant_code_file"]
+                and solution_item["relevant_code_file"]["repo_name"]
+            ):
+                repos.add(solution_item["relevant_code_file"]["repo_name"])
+
+        return list(repos)
+
+    def _launch_agents_for_repos(
+        self,
+        installation: CodingAgentIntegration,
+        autofix_state: AutofixState,
+        run_id: int,
+        organization,
+        trigger_source: AutofixTriggerSource,
+    ) -> list[dict]:
+        """Launch coding agents for all repositories in the solution."""
+        repos = self._extract_repos_from_solution(autofix_state)
+
+        if not repos:
+            logger.error(
+                "coding_agent.post.no_repos_found_in_solution",
+                extra={
+                    "organization_id": organization.id,
+                    "run_id": run_id,
+                },
+            )
+
+            # Fallback to run on all repos, specific to github's setup for now as our coding agent integrations only support github.
+            repos = [f"{repo.owner}/{repo.name}" for repo in autofix_state.request["repos"]]
+
+            if not repos:
+                raise NotFound("No repos to run agents")
+
+        prompt = get_coding_agent_prompt(run_id, trigger_source)
+
+        if not prompt:
+            raise APIException("No prompt to send to agents.")
+
+        results = []
+
+        for repo_name in repos:
+            try:
+                repo = next(
+                    (
+                        repo
+                        for repo in autofix_state.request["repos"]
+                        if f"{repo.owner}/{repo.name}" == repo_name
+                    ),
+                    None,
+                )
+                if not repo:
+                    logger.error(
+                        "coding_agent.repo_not_found",
+                        extra={
+                            "organization_id": organization.id,
+                            "run_id": run_id,
+                            "repo_name": repo_name,
+                        },
+                    )
+                    # Continue with other repos instead of failing entirely
+                    continue
+
+                launch_request = CodingAgentLaunchRequest(
+                    prompt=prompt,
+                    repository=repo,
+                    branch_name=sanitize_branch_name(autofix_state.request["issue"]["title"]),
+                )
+
+                coding_agent_state = installation.launch(launch_request)
+
+                store_coding_agent_state_to_seer(
+                    run_id=run_id,
+                    coding_agent_state=coding_agent_state,
+                )
+
+                results.append(
+                    {
+                        "repo_name": repo_name,
+                        "coding_agent_state": coding_agent_state,
+                    }
+                )
+
+            except Exception:
```

Comment:
what are we expecting to fail and why? can we be more surgical in our failure handling?